streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -134,9 +134,7 @@ private static FileLock lockStateDirectory(File stateDir, int retry) throws IOEx
             retry--;
             lock = lockStateDirectory(channel);
         }
-        if (lock == null) {
-            channel.close();
-        }
+
         return lock;
```

Comment:
Also, can you please explain why we don't close the channel even if we don't successfully acquire a lock? It may be worth a comment in the code too. I assume